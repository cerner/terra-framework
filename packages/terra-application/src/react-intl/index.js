export { default } from 'react-intl';
export * from 'react-intl';
