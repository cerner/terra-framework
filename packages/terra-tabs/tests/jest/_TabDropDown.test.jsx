/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { IntlProvider } from 'react-intl';
import { mountWithIntl } from 'terra-enzyme-intl';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import TabDropDown from '../../src/common-tabs/_TabDropDown';
import HiddenTab from '../../src/common-tabs/_HiddenTab';

describe('TabDropDown', () => {
  test('should render the dropdown with provided hidden tabs', () => {
    const mockOnSelect = jest.fn();
    let mockRef;
    render((
      <IntlProvider>
        <TabDropDown
          refCallback={(ref) => { mockRef = ref; }}
        >
          <HiddenTab
            id="tab-1"
            associatedPanelId="panel-1"
            index={0}
            label="Tab 1 Label"
            itemKey="tab-1-key"
            onSelect={mockOnSelect}
            tabIds={['tab-1', 'tab-2']}
            onBlur={() => {}}
            onFocus={() => {}}
          />
          <HiddenTab
            id="tab-2"
            associatedPanelId="panel-2"
            index={0}
            label="Tab 2 Label"
            itemKey="tab-2-key"
            onSelect={mockOnSelect}
            tabIds={['tab-1', 'tab-2']}
            onBlur={() => {}}
            onFocus={() => {}}
          />
        </TabDropDown>
      </IntlProvider>
    ));

    expect(screen.getByRole('none')).toBe(mockRef);
    expect(screen.getByRole('tab', { name: 'Tab 1 Label' })).toBeInTheDocument();
    expect(screen.getByRole('tab', { name: 'Tab 2 Label' })).toBeInTheDocument();

    const wrapper = mountWithIntl(
      <IntlProvider>
        <TabDropDown
          refCallback={(ref) => { mockRef = ref; }}
        >
          <HiddenTab
            id="tab-1"
            associatedPanelId="panel-1"
            index={0}
            label="Tab 1 Label"
            itemKey="tab-1-key"
            onSelect={mockOnSelect}
            tabIds={['tab-1', 'tab-2']}
            onBlur={() => {}}
            onFocus={() => {}}
          />
          <HiddenTab
            id="tab-2"
            associatedPanelId="panel-2"
            index={0}
            label="Tab 2 Label"
            itemKey="tab-2-key"
            onSelect={mockOnSelect}
            tabIds={['tab-1', 'tab-2']}
            onBlur={() => {}}
            onFocus={() => {}}
          />
        </TabDropDown>
      </IntlProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
