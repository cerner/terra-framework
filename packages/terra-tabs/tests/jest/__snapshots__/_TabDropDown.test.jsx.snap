// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabDropDown should render the dropdown with provided hidden tabs 1`] = `
<IntlProvider
  intl={
    {
      "defaultFormats": {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": {},
      "formatters": {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <OnClickOutside(InjectIntl(TabDropDown))
    eventTypes={
      [
        "mousedown",
        "touchstart",
      ]
    }
    excludeScrollbar={false}
    outsideClickIgnoreClass="ignore-react-onclickoutside"
    preventDefault={false}
    refCallback={[Function]}
    stopPropagation={false}
  >
    <InjectIntl(TabDropDown)
      disableOnClickOutside={[Function]}
      enableOnClickOutside={[Function]}
      eventTypes={
        [
          "mousedown",
          "touchstart",
        ]
      }
      outsideClickIgnoreClass="ignore-react-onclickoutside"
      preventDefault={false}
      refCallback={[Function]}
      stopPropagation={false}
    >
      <TabDropDown
        disableOnClickOutside={[Function]}
        enableOnClickOutside={[Function]}
        eventTypes={
          [
            "mousedown",
            "touchstart",
          ]
        }
        intl={
          {
            "defaultFormats": {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": {},
            "formatters": {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": null,
            "now": [Function],
            "onError": [Function],
            "textComponent": "span",
            "timeZone": null,
          }
        }
        outsideClickIgnoreClass="ignore-react-onclickoutside"
        preventDefault={false}
        refCallback={[Function]}
        stopPropagation={false}
      >
        <div
          className="drop-down"
          onMouseDown={[Function]}
          role="none"
        >
          <InjectIntl(HiddenTab)
            associatedPanelId="panel-1"
            id="tab-1"
            index={0}
            itemKey="tab-1-key"
            label="Tab 1 Label"
            onBlur={[Function]}
            onFocus={[Function]}
            onSelect={[MockFunction]}
            tabIds={
              [
                "tab-1",
                "tab-2",
              ]
            }
          >
            <HiddenTab
              associatedPanelId="panel-1"
              id="tab-1"
              index={0}
              intl={
                {
                  "defaultFormats": {},
                  "defaultLocale": "en",
                  "formatDate": [Function],
                  "formatHTMLMessage": [Function],
                  "formatMessage": [Function],
                  "formatNumber": [Function],
                  "formatPlural": [Function],
                  "formatRelative": [Function],
                  "formatTime": [Function],
                  "formats": {},
                  "formatters": {
                    "getDateTimeFormat": [Function],
                    "getMessageFormat": [Function],
                    "getNumberFormat": [Function],
                    "getPluralFormat": [Function],
                    "getRelativeFormat": [Function],
                  },
                  "locale": "en",
                  "messages": null,
                  "now": [Function],
                  "onError": [Function],
                  "textComponent": "span",
                  "timeZone": null,
                }
              }
              isClosable={false}
              isDisabled={false}
              isDraggable={false}
              isSelected={false}
              itemKey="tab-1-key"
              label="Tab 1 Label"
              onBlur={[Function]}
              onFocus={[Function]}
              onSelect={[MockFunction]}
              showAddButton={false}
              showIcon={false}
              tabIds={
                [
                  "tab-1",
                  "tab-2",
                ]
              }
            >
              <div
                aria-controls="panel-1"
                aria-disabled={false}
                aria-selected={false}
                className="hidden"
                data-focus-styles-enabled={true}
                id="tab-1"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="tab"
                tabIndex={-1}
              >
                <VisuallyHiddenText
                  aria-hidden={true}
                  id="terra-hidden-tab-pane-response=00000000-0000-0000-0000-000000000000"
                  text="Terra.tabs.addbutton.focus"
                >
                  <span
                    aria-hidden={true}
                    className="visually-hidden-text"
                    id="terra-hidden-tab-pane-response=00000000-0000-0000-0000-000000000000"
                  >
                    Terra.tabs.addbutton.focus
                  </span>
                </VisuallyHiddenText>
                <VisuallyHiddenText
                  aria-hidden={true}
                  id="terra-tab-close-response=00000000-0000-0000-0000-000000000000"
                  text="Terra.tabs.hint.removable"
                >
                  <span
                    aria-hidden={true}
                    className="visually-hidden-text"
                    id="terra-tab-close-response=00000000-0000-0000-0000-000000000000"
                  >
                    Terra.tabs.hint.removable
                  </span>
                </VisuallyHiddenText>
                <div
                  className="checkbox"
                />
                <div
                  className="label"
                >
                  Tab 1 Label
                </div>
              </div>
            </HiddenTab>
          </InjectIntl(HiddenTab)>
          <InjectIntl(HiddenTab)
            associatedPanelId="panel-2"
            id="tab-2"
            index={0}
            itemKey="tab-2-key"
            label="Tab 2 Label"
            onBlur={[Function]}
            onFocus={[Function]}
            onSelect={[MockFunction]}
            tabIds={
              [
                "tab-1",
                "tab-2",
              ]
            }
          >
            <HiddenTab
              associatedPanelId="panel-2"
              id="tab-2"
              index={0}
              intl={
                {
                  "defaultFormats": {},
                  "defaultLocale": "en",
                  "formatDate": [Function],
                  "formatHTMLMessage": [Function],
                  "formatMessage": [Function],
                  "formatNumber": [Function],
                  "formatPlural": [Function],
                  "formatRelative": [Function],
                  "formatTime": [Function],
                  "formats": {},
                  "formatters": {
                    "getDateTimeFormat": [Function],
                    "getMessageFormat": [Function],
                    "getNumberFormat": [Function],
                    "getPluralFormat": [Function],
                    "getRelativeFormat": [Function],
                  },
                  "locale": "en",
                  "messages": null,
                  "now": [Function],
                  "onError": [Function],
                  "textComponent": "span",
                  "timeZone": null,
                }
              }
              isClosable={false}
              isDisabled={false}
              isDraggable={false}
              isSelected={false}
              itemKey="tab-2-key"
              label="Tab 2 Label"
              onBlur={[Function]}
              onFocus={[Function]}
              onSelect={[MockFunction]}
              showAddButton={false}
              showIcon={false}
              tabIds={
                [
                  "tab-1",
                  "tab-2",
                ]
              }
            >
              <div
                aria-controls="panel-2"
                aria-disabled={false}
                aria-selected={false}
                className="hidden"
                data-focus-styles-enabled={true}
                id="tab-2"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="tab"
                tabIndex={-1}
              >
                <VisuallyHiddenText
                  aria-hidden={true}
                  id="terra-hidden-tab-pane-response=00000000-0000-0000-0000-000000000000"
                  text="Terra.tabs.addbutton.focus"
                >
                  <span
                    aria-hidden={true}
                    className="visually-hidden-text"
                    id="terra-hidden-tab-pane-response=00000000-0000-0000-0000-000000000000"
                  >
                    Terra.tabs.addbutton.focus
                  </span>
                </VisuallyHiddenText>
                <VisuallyHiddenText
                  aria-hidden={true}
                  id="terra-tab-close-response=00000000-0000-0000-0000-000000000000"
                  text="Terra.tabs.hint.removable"
                >
                  <span
                    aria-hidden={true}
                    className="visually-hidden-text"
                    id="terra-tab-close-response=00000000-0000-0000-0000-000000000000"
                  >
                    Terra.tabs.hint.removable
                  </span>
                </VisuallyHiddenText>
                <div
                  className="checkbox"
                />
                <div
                  className="label"
                >
                  Tab 2 Label
                </div>
              </div>
            </HiddenTab>
          </InjectIntl(HiddenTab)>
        </div>
      </TabDropDown>
    </InjectIntl(TabDropDown)>
  </OnClickOutside(InjectIntl(TabDropDown))>
</IntlProvider>
`;
